app/helpers/home_teacher_helper.rb

```diff
@@ -90,14 +90,140 @@ def calculate_objective_width(course)
   end
 
   def calculate_assessment_width(course)
-    na = number_of_assessments(course)
-    units = na + 4 + 3 + 6 # assessments + objectives + last_activity + names
+    # na = number_of_assessments(course)
+    # units = na + 4 + 3 + 6 # assessments + objectives + last_activity + names
     # 1.0 / units * 100
     4.0
   end
 
   def objective_summary(uid, course)
-    '<td></td><td></td><td></td><td></td>'.html_safe
+    os = collect_objective_info(uid, course).map do |oi|
+      {
+        objective: summarize_objective(uid, oi)
+      }
+    end
+    format_objective_summary(os)
+  end
+
+  def collect_objective_info(uid, course)
+    Course.find(course[:course]).objectives.map do |o|
+      {
+        info: o,
+        strands: o.strands.map do |s|
+          {
+            info: s,
+            assessments: s.assessments.where(section: course[:section]).map do |a|
+              {
+                assessment: a,
+                scores: Score.where(user: uid, strand: s, assessment: a)
+              }
+            end
+          }
+        end
+      }
+    end
+  end
+
+  def format_objective_summary(os)
+    os.map do |o|
+      # For each objective.
+      score = 0.0
+      scored_strands = 0.0
+      unless objective_has_assessments(o) == true
+        css_class = '\'not-assessed-yet\''
+        score = 'NA'
+        # tooltip = "No assessments yet for this objective."
+      else
+        o[:objective][:strands].map do |s|
+          # For each strand.
+          unless s[:score][:roll_up_score].nil?
+            score += s[:score][:roll_up_score].to_f
+            scored_strands += 1
+          end
+          scored_strands > 0 ? score = score / scored_strands : 0.0
+        end
+        css_class = case
+                    when score <= 1.0
+                      'limited-knowledge'
+                    when score <= 2.0
+                      'adequate-knowledge'
+                    when score <  6.0
+                      'substantial-knowledge'
+                    else
+                      'excellent-knowledge'
+                    end
+      end
+      "<td class=#{css_class}><strong>#{score}</strong></td>"
+    end.join.html_safe
+  end
+
+  def objective_has_assessments(o)
+    if o[:objective][:strands].map { |st| st[:score][:assessments] }.sum == 0
+      false
+    else
+      true
+    end
+  end
+
+  def summarize_objective(uid, objective)
+    {
+      info: objective[:info],
+      strands: objective[:strands].map do |s|
+        {
+          info: s[:info],
+          score: strand_score(uid, s)
+        }
+      end
+    }
+  end
+
+  def strand_score(uid, s)
+    {
+      assessments: s[:assessments].count,
+      posts: count_posts_for_strand(uid, s),
+      scores: times_assessed(s),
+      max_possible_score: max_possible_score(s),
+      raw_score: raw_score(s),
+      roll_up_method: 'roll_up_average',
```

Comment:
Making `roll_up_average` a symbol (`:roll_up_average`) would be more idiomatic Ruby.